mozilla-central/setup

```diff
@@ -46,7 +49,102 @@ pushd $GIT_ROOT
+# repo, and we don't have the source corresponding to the indexing run on taskcluster.
+# In that case we should just build the latest version of gecko-dev that we do have,
+# so that we have analysis data for a version of the source we have.
+# Whether or not we do the build is decided in the "build" script in the same folder
+# as this file, which detects this scenario by the absence of the
+# target.mozsearch-index.zip file.
+
+if [ -n "$INDEXED_GIT_REV" ]; then
+    pushd $GIT_ROOT
+    # Since we're using a prebuilt index at a particular rev, sync our local repo to the same rev
+    git checkout $INDEXED_GIT_REV
+    popd
+
+    # Download the C++ analysis, rust save-analysis files, and generated sources tarballs.
+
+    rm -rf $INDEX_ROOT/analysis
+    rm -rf $INDEX_ROOT/objdir
+
+    rm -f target.mozsearch-index.zip
+    wget -nv https://index.taskcluster.net/v1/task/gecko.v2.$REVISION.firefox.linux64-searchfox-debug/artifacts/public/build/target.mozsearch-index.zip
+    mkdir -p $INDEX_ROOT/analysis
+    unzip target.mozsearch-index.zip -d $INDEX_ROOT/analysis
+
+    rm -f target.mozsearch-rust.zip
+    wget -nv https://index.taskcluster.net/v1/task/gecko.v2.$REVISION.firefox.linux64-searchfox-debug/artifacts/public/build/target.mozsearch-rust.zip
+    mkdir -p $INDEX_ROOT/objdir
+    unzip target.mozsearch-rust.zip -d $INDEX_ROOT/objdir
+
+    rm -f target.generated-files.tar.gz
+    wget -nv https://index.taskcluster.net/v1/task/gecko.v2.$REVISION.firefox.linux64-searchfox-debug/artifacts/public/build/target.generated-files.tar.gz
+    tar -x -z -C $INDEX_ROOT/objdir -f target.generated-files.tar.gz
+
+    # Special cases - buildid.h and mozilla-config.h show up in two places in a regular
+    # objdir and the analysis tarball will correspondingly also have two instances of
+    # the analysis file. The generated-files tarball only has one copy, so we manually
+    # make the other copy to make things match up. The best would be if the gecko build
+    # system didn't actually produce two copies of these files.
+    cp $INDEX_ROOT/objdir/buildid.h $INDEX_ROOT/objdir/dist/include/buildid.h
+    cp $INDEX_ROOT/objdir/mozilla-config.h $INDEX_ROOT/objdir/dist/include/mozilla-config.h
+
+    # We get analysis data for generated files, some of which aren't included
+    # in the generated-files tarball that we get from taskcluster. Most of these
+    # missing files are dummy unified build files that we don't care about, and
+    # we can just delete those.
+    # If there are other such cases, then we'll get zero-byte files generated by
+    # output-file.rs since it won't be able to find the source file corresponding to
+    # the analysis file. In those cases we should ensure the generated file is
+    # included in the target.generated-files.tar.gz tarball that comes out of the
+    # taskcluster indexing job. Bug 1440879 can be used as a guide.
+    pushd $INDEX_ROOT/analysis/__GENERATED__
+    find . -type f -name "Unified*" |
```

Comment:
It's a bit unfortunate to hardcode this, but I guess the build system doesn't have a list of unified objects? Looks like it shouldn't be too hard to generate such a list, maybe worth filing a bug?

Reply:
I don't know if there's much benefit to doing that. I could file a bug but chances are it will never get fixed. A better change might be to strip out the analysis files for the Unified* files from the analysis tarball.